notificationsender/firebase.js

```diff
@@ -1,58 +1,54 @@
+const firebase = require('firebase-admin');
 const logger = require('../logger.js');
 const redis = require('../redis-helper');
 
-const firebaseClient = new Firebase(system.getGcmPassword());
+if (system.isGcmConfigured()) {
+    const serviceAccount = require(system.getFirebaseServiceFile());
+    firebase.initializeApp({
+        credential: firebase.credential.cert(serviceAccount)
+    });
+}
 
-const firebaseOptions = {
-    delay_while_idle: false,
-    priority: 'high'
+function sendMessage(registrationIds, data) {
+    const message = {
+        data: data,
+        tokens: Array.isArray(registrationIds) ? registrationIds : [registrationIds],
+        android: {
+            priority: 'high',
+        }
+    };
+    firebase.messaging().sendMulticast(message)
+        .then((response) => {
+            logger.info("Response: " + JSON.stringify(response));
+        })
+        .catch(error => {
+            logger.error("GCM send error: ", error);
+        });
 };
 
-function sendNotificationWithData(registrationIds, data) {
-    redis.incr("androidNotificationId", function(error, androidNotificationId) {
+exports.sendNotification = function (registrationIds, notification) {
+    redis.incr("androidNotificationId", function (error, androidNotificationId) {
         if (error) {
             return;
         }
-
-        data.type = 'notification';
-        data.notificationId = androidNotificationId;
-        firebaseClient.message(registrationIds, data, firebaseOptions, function (result) {
-            if (result.failure) {
-                logger.error("GCM send error: " + JSON.stringify(result));
-            }
-        });
+        const data = {
+            message: notification.message,
+            type: 'notification',
+            severity: notification.severity || '',
+            icon: notification.icon || '',
+            persistedId: notification._id.toString(),
+            timestamp: notification.created.getTime().toString(),
+            notificationId: androidNotificationId.toString()
+        };
+        sendMessage(registrationIds, data);
     });
 };
 
-exports.sendMessageNotification = function(registrationIds, message) {
-    var data = {
-        message: message,
-        timestamp: Date.now()
-    };
-    sendNotificationWithData(registrationIds, data);
-};
-
-exports.sendNotification = function(registrationIds, notification) {
-    var data = {
-        message: notification.message,
-        severity: notification.severity,
-        icon: notification.icon,
-        persistedId: notification._id,
-        timestamp: notification.created.getTime()
-    };
-    sendNotificationWithData(registrationIds, data);
-};
-
-exports.hideNotification = function(registrationIds, notificationId) {
+exports.hideNotification = function (registrationIds, notificationId) {
     const data = {
         type: 'hideNotification',
-        notificationId: notificationId
+        notificationId: notificationId.toString()
```

Comment:
Maybe migrate this to `persistedId`?

Reply:
It is already: https://github.com/openhab/openhab-cloud/pull/449/files#diff-4E8B28A0F7943CB0FE11996DD13FD7DFR56 ... Just the parameter needs renaming.

Reply:
After further thought, were you just referring to the key name? If so, I'd agree.